### DISCUSSION TITLE
Show uncompressed at deepest level ?

### DISCUSSION THREAD

**Discussion by seafjordking** [openseadragon/openseadragon#2330] [Q&A]
Hi,

A newbie to this, so probably a stupid question:
- I have 16bit uncompressed tiff as the original file(s)
- The image pyramid produced will be a range of 8bit PNG or JPEG files (if I understand it correctly)

Would it be possible to make the lower zoom levels to remain 8bit PNG/JPEG, but allow the deepest level (largest zoom) to show uncompressed 16bit tif?
I loose some high-res information when compressed, which is a bit cumbersome. 
I don't mind that the lower zoom levels are compressed, but when zooming in at the highest zoom, I would like that to show the uncompressed, original 16bit tiff.

**Comment by iangilman** [comment]
Seems like a reasonable question! 

OSD uses the browser's 2D canvas, which, as far as I know, can only display 8bit images. Even if you had 16bit data, you'd have to convert it to 8bit before it was displayed.

OSD doesn't support mixing file types in a single tile pyramid. If you want to make sure your lowest level is uncompressed, you should use PNG, which is lossless. It'll be a little heavier than JPEG, but not too bad. 

There are plans underway to be able to draw using WebGL, which would open the door to displaying 16bit images. You can follow along on that in https://github.com/openseadragon/openseadragon/pull/2310 if you're interested. It probably won't be ready until the fall, though. 

If you really want to go with JPEG for the lower levels and 16bit TIFF for the highest level, you could create a custom data transport layer using this API: 

https://openseadragon.github.io/examples/advanced-data-model/

You would, of course, need to convert the 16bit data to 8bit before giving it to OSD to draw, however. 

I hope this helps!
